### DISCUSSION TITLE
Fresh helm deployment throws redirect_uri error

### DISCUSSION THREAD

**Discussion by shedar** [zitadel/zitadel#4048] [Q&A] [answered]
Hello,

I'm trying to deploy zitadel using helm chart via terraform.
But when I try to sign in to the console - I get an error message: `The requested redirect_uri is missing in the client configuration. If you have any questions, you may contact the administrator of the application.`

`/ui/console/assets/environment.json` returns  a valid config at first glance:
```json
{"api":"https://my.domain","issuer":"https://my.domain","clientid":"171675571183559858@zitadel"}
```

My current terraform config:
```hcl
resource "helm_release" "zitadel" {
  depends_on   = [ kubernetes_namespace.zitadel ]
  namespace    = var.zitadel_namespace
  name         = "zitadel"
  chart        = "zitadel"
  version      = "2.2.2"

  repository   = "https://charts.zitadel.com"
  # wait=false is needed for cocroachdb
  # https://www.cockroachlabs.com/docs/stable/deploy-cockroachdb-with-kubernetes.html?filters=helm
  # remove it after zitadel supports postgresql
  wait = false
  values = [yamlencode({
    image = { tag = "v2.0.0-v2-alpha.43-arm64"}
    zitadel = {
      masterkey = random_password.zitadel_master.result
      secretConfig = {
        Database = {
          User = {
            Password = random_password.zitadel_db_password.result
          }
        }
      }
      configmapConfig = {
        ExternalDomain = var.zitadel_domain
        S3CustomDomain = var.zitadel_domain
        S3DefaultInstance = {
          CustomDomain = var.zitadel_domain
        }
        ExternalSecure = true
        TLS = {Enabled = false}
      }
    }
    replicaCount = 1
    ingress = {
      enabled   = true,
      className = "nginx"
      annotations= {
        "cert-manager.io/cluster-issuer"= "letsencrypt"
      }
      hosts = [
        {
          host = var.zitadel_domain,
          paths: [{
            path = "/",
            pathType = "ImplementationSpecific"
          }]
        }
      ],
      tls = [
        {
          secretName = "zitadel-tls",
          hosts = [var.zitadel_domain]
        }
      ]
    }


  })]
}

```

**Comment by fforootd** [comment, marked as answer]
Hi - @shedar 

This sounds to me like a problem with the tls mode. TLS mode docs are a WIP here #4046 
It looks like you run an nginx in front of ZITADEL and connect from nginx to ZITADEL without TLS, true?
In that case you need to add the ` --tlsMode disabled` to ZITADEL args.

Can you share the rendered deployments `arg` for the container?

**Reply by fforootd** [reply]
@eliobischof what does the chart use as default for the tls mode?

**Reply by eliobischof** [reply]
Yes, the charts defaults are not updated for the new TLS config yet. This will change very soon with the release of v2.3.0.

In the meantime, you can set `zitadel.configmapConfig.TLS.Enabled` to `false` and `zitadel.configmapConfig.ExternalSecure` to `false` for HTTP or `true` for HTTPS through a TLS terminating proxy

**Reply by eliobischof** [reply]
I see, you already have that configured.
Instead, I think you need to configure `zitadel.configmapConfig.ExternalPort=443`.

BTW: S3CustomDomain is not a known configuration key, you can delete that. We had that wrong in a recent chart README, I think.

**Reply by shedar** [reply]
@eliobischof Adding `zitadel.configmapConfig.ExternalPort=443` solved the issue. Thank you for the help!

As an idea, maybe it is worth implicitly setting the default value of the external port to 443 with "ExternalSecure=true"

**Reply by fforootd** [reply]
Awesome. Let us know if any other problems arise.
